docs/examples/oceanmodelling.py

```diff
@@ -0,0 +1,509 @@
+#
+# $$
+# \frac{\partial}{\partial x^{(z)}} = - \frac{\partial}{\partial \left( x^\prime \right)^{(z)}}
+# $$
+#
+# for either $z$.
+# %%
+@dataclass
+class HelmholtzKernel(gpx.kernels.AbstractKernel):
+    # initialise Phi and Psi kernels as any stationary kernel in gpJax
+    potential_kernel: gpx.kernels.AbstractKernel = gpx.kernels.RBF(active_dims=[0, 1])
+    stream_kernel: gpx.kernels.AbstractKernel = gpx.kernels.RBF(active_dims=[0, 1])
+
+    def __call__(
+        self, X: Float[Array, "1 D"], Xp: Float[Array, "1 D"]
+    ) -> Float[Array, "1"]:
+        # obtain indices for k_helm, implement in the correct sign between the derivatives
+        z = jnp.array(X[2], dtype=int)
+        zp = jnp.array(Xp[2], dtype=int)
+        sign = (-1) ** (z + zp)
+
+        # convert to array to correctly index, -ve sign due to exchange symmetry (only true for stationary kernels)
+        potential_dvtve = -jnp.array(
+            hessian(self.potential_kernel)(X, Xp), dtype=jnp.float64
+        )[z][zp]
+        stream_dvtve = -jnp.array(
+            hessian(self.stream_kernel)(X, Xp), dtype=jnp.float64
+        )[1 - z][1 - zp]
+
+        return potential_dvtve + sign * stream_dvtve
+
+
+# %% [markdown]
+# ### GPJax Implementation
+# We repeat the exact same steps as with the velocity GP model, but replacing `VelocityKernel` with `HelmholtzKernel`.
+
+# %%
+# Redefine Gaussian process with Helmholtz kernel
+kernel = HelmholtzKernel()
+helmholtz_posterior = initialise_gp(kernel, mean, dataset_train)
+# Optimise hyperparameters using optax
+opt_helmholtz_posterior = optimise_mll(helmholtz_posterior, dataset_train)
+
+
+# %% [markdown]
+# ### Comparison
+# We again plot the ground truth (testing data) $D_0$, the predicted latent vector field $\mathbf{F}_{\text{latent}}(\mathbf{x_i})$, and a heatmap of the residuals at each location $R(\mathbf{x}_i) = \mathbf{y}_{0,i} - \mathbf{F}_{\text{latent}}(\mathbf{x}_i)$ and $\left|\left|R(\mathbf{x}_i)  \right|\right|$.
+
+# %%
+# obtain latent distribution, extract x and y values over g
+helmholtz_mean, helmholtz_std = latent_distribution(
+    opt_helmholtz_posterior, dataset_ground_truth.X, dataset_train
+)
+dataset_latent_helmholtz = dataset_3d(pos_test, helmholtz_mean)
+
+plot_fields(dataset_ground_truth, dataset_train, dataset_latent_helmholtz)
+
+# %% [markdown]
+# Visually, the Helmholtz model performs better than the velocity model, preserving the local structure of the $\mathbf{F}$. Since we placed priors on $\Phi$ and $\Psi$, the construction of $\mathbf{F}$ allows for correlations between the dimensions (non-zero off diagonal elements in the Gram matrix populated by $k_\text{Helm}\left(\mathbf{X},\mathbf{X}^{\prime}\right)$ ).
+
```

Comment:
```suggestion
# Visually, the Helmholtz model performs better than the velocity model, preserving the local structure of the $\mathbf{F}$. Since we placed priors on $\Phi$ and $\Psi$, the construction of $\mathbf{F}$ allows for correlations between the dimensions (non-zero off-diagonal elements in the Gram matrix populated by $k_\text{Helm}\left(\mathbf{X},\mathbf{X}^{\prime}\right)$ ).

```